### PR TITLE
Run Github Workflows On All Branches

### LINKED ISSUE #1099: Run Github Workflows On All Branches
### Issue Description

Currently Github workflows don't run on all branches.

### Severity/Priority
 - [ ] Critical
 - [ ] High
 - [x] Medium
 - [ ] Low

### Checklist

- [ ] I have read and followed the project's code of conduct.
- [ ] I have searched for similar issues before creating this one.
- [ ] I have provided all the necessary information to understand and reproduce the issue.
- [ ] I am willing to contribute to the resolution of this issue.

### PR DESCRIPTION
**Pull request by ajoykumardas12**
Date: 31th Jan 2024

Developer Name: @ajoykumardas12 

---

## Issue Ticket Number

closes #1099 

## Description
Updated Github workflows to run on all branches. Related Docs: [Link](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#patterns-to-match-branches-and-tags)

Previously branches which had `/` in name were ignored from workflows. eg, #1098 

- Upgraded actions/checkout to v3 as v2 is being deprecated. Also upgraded actions/setup-node to v3.

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots

No visual change.

## Test Coverage

This PR doesn't affect tests on dev, affects tests on Github CI.

### CODE REVIEW

File: .github/workflows/continuous-integration.yml
Comment (ajoykumardas12) [inline .github/workflows/continuous-integration.yml:1]: Only change is `'*'` -> `'**'`, rest are from prettier formatting of this repo.

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -4,26 +4,25 @@
...
-      uses: actions/setup-node@v1
-      with:
-        node-version: ${{ matrix.node-version }}
-    - run: yarn
-    - run: yarn lint
-    - run: yarn prettier --check
-    - run: yarn test
-    - run: yarn build
+        steps:
+            - uses: actions/checkout@v2
```
Comment (prakashchoudhary07) [inline .github/workflows/continuous-integration.yml:19]: With this can you also update the action/checkout to v3,
I think v2 was being depricated

Comment (ajoykumardas12) [reply .github/workflows/continuous-integration.yml]: Done.
Also updated actions/setup-node to v3

**Comment by sahsisunny** [comment]
It looks good to me, but I suggest creating a separate workflow for running tests that run on all the branches. For the remaining tasks such as building, checking, and Prettier, it would be beneficial to create separate workflows that run specifically on the `develop` and `main` branches.

**Review by sahsisunny — APPROVED** [review]

**Comment by prakashchoudhary07** [comment]
> It looks good to me, but I suggest creating a separate workflow for running tests that run on all the branches. For the remaining tasks such as building, checking, and Prettier, it would be beneficial to create separate workflows that run specifically on the `develop` and `main` branches.

I think one should know if checks passed, even before creating a PR to develop branch

**Review by prakashchoudhary07 — APPROVED** [review]

**Comment by ajoykumardas12** [comment]
Also, GitHub workflow is free for public repos, so won't have any issues [Workflow limits](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#about-billing-for-github-actions)

**Pull request merged**